### ISSUE TITLE
UnusedImport false positive when the companion object of the target is referenced

### ISSUE THREAD

**Issue by matejdro** [detekt/detekt#8989]
## Steps to Reproduce

Run detekt on the following code:

```kotlin
import kotlinx.coroutines.CoroutineDispatcher

fun main() {
    println(CoroutineDispatcher)
}
```

## Expected Behavior

There should be no unused import reports 

## Observed Behavior

Detekt reports `Main.kt:1:1: The import 'kotlinx.coroutines.CoroutineDispatcher' is unused. [UnusedImport]`

## Your Environment

* Version of detekt used: 2.0.0-alpha.1
